logic/src/commonMain/kotlin/com/wire/kalium/logic/data/event/Event.kt

```diff
@@ -83,14 +83,28 @@ data class EventEnvelope(
  * @property source The source of the event.
  * @see EventSource
  */
-data class EventDeliveryInfo(
-    val isTransient: Boolean,
-    val source: EventSource,
+sealed class EventDeliveryInfo(
+    open val isTransient: Boolean,
+    open val source: EventSource
 ) {
+
     fun toLogMap(): Map<String, Any?> = mapOf(
         "isTransient" to isTransient,
         "source" to source.name
     )
+
+    data class AsyncEventDeliveryInfo(
```

Comment:
Nit:

```suggestion
    data class Async(
```

It's gonna look like `EventDeliveryInfo.Async`, instead of `EventDeliveryInfo.AsyncEventDeliveryInfo`, which is kinda redundant.